apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/exception/Exceptions.kt

```diff
@@ -6,10 +6,18 @@ package com.apollographql.apollo3.exception
 open class ApolloException(message: String? = null, cause: Throwable? = null) : RuntimeException(message, cause)
 
 /**
- * A network happened: socket closed, DNS issue, TLS problem, etc...
+ * A network error happened: socket closed, DNS issue, TLS problem, etc...
  */
 class ApolloNetworkException(message: String? = null, cause: Throwable? = null) : ApolloException(message = message, cause = cause)
 
+/**
+ * A WebSocket connection could not be established: e.g., expired token
+ */
+class ApolloWebSocketException(
```

Comment:
Do you need a separate exception for this given that there is `ApolloNetworkException`?

Reply:
I thought about using ApolloNetworkException but then I would have to hack how the code would be included. So, I decided to introduce an explicit exception.

Reply:
What about `ApolloWebSocketClosedException` ? This leaves the door open for potentially other `WebSocketExceptions` in the future if we need to? I think the code is only there when closing the websocket?

Reply:
ApolloWebSocketClosedException sounds good to me.